Driver for Becker AR62xx radios

eku60 opened the pull request:
Because of problems with my first pull request, i open a new one which should be better to understand.

MaxKellermann commented:
Looks just like #189 ...
Please check the CircleCI build failure.

MaxKellermann commented:
Note that you don't need to create a new PR - just force-push over the existing branch to update this PR.

eku60 commented:
Hello Max,

I'm still learning GIT. I found SVN easier to handle.

I have read the log (720 pages! when opening in Ocular) and will correct the items (LogFormat(...) for example).
Then the next try will start.

MaxKellermann commented:
Code reviews are not possible with Subversion, that's why it's easier for you. You would just have committed those broken changes, just like you did with git (the process of committing code is nearly identical). The difference is that you only messed with your private copy, not with the official repository.
Having to clean up your work *before* breaking everything for everybody may seem cumbersome. But it's the right thing to do.

MaxKellermann commented:
Your last commit (804630e893698513dbfeb11f2b83bd1035e1100d) looks completely wrong. I have no idea what you're doing.

eku60 commented:
Hallo Max,
 
ich habe irgendwie Probleme mit Git im Eclipse und kämpfe damit.
Eigentlich wollte ich 4 Dateien (im Anhang) committen, komme aber nicht an 
meinen fork "eku60/XCSoar" dran. Irgendwas ist bei mir noch falsch 
eingestellt. Ich bin auf der Suche.
 
Bis es soweit ist, dass das alles bei mir wie gewünscht funktioniert, sende 
ich Dir mal die 4 Dateien hier zu, damit XCSoar über einen Treiber für das 
Becker AR6201 verfügt. Den werden neben mir sicher viele sehnlichst 
erwarten.
 
Mit Hilfe von Imker habe ich jetzt auch ein git diff hingekriegt. Das hänge 
ich hier ebenfalls dran (.patch). Du kannst den Treiber dann gerne 
implementieren. Würde mich freuen. Ich verbrauche die 5 fache Zeit für den 
Ärger mit der Umgebung als fürs eigentliche Problem, nämlich den Treiber.
Mit TARGET=UNIX und PC oder WIN64 habe ich XCSoar kompiliert bekommen und 
dann den Treiber an einem Funkgerät an der seriellen Schnittstelle 
getestet. Das hat funktioniert.
 
Viele Grüße
Elmar Kümmel
eku60
 
 
 
 
 
-----Original-Nachricht-----
Betreff: Re: [XCSoar/XCSoar] Driver for Becker AR62xx radios (#190)
Datum: 2019-05-16T21:40:31+0200
Von: "Max Kellermann" <notifications@github.com>
An: "XCSoar/XCSoar" <XCSoar@noreply.github.com>
 
 
 
Your last commit (804630e
<https://github.com/XCSoar/XCSoar/commit/804630e893698513dbfeb11f2b83bd1035e1100d> 
) looks completely wrong. I have no idea what you're doing.
—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/XCSoar/XCSoar/pull/190?email_source=notifications&email_token=AL7IIITM4U6E43VQU3ZR3CLPVW2CRA5CNFSM4HKYTEA2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODVS3KNY#issuecomment-493204791> 
, or mute the thread
<https://github.com/notifications/unsubscribe-auth/AL7IIIWD3W6VXOFHAQLWFP3PVW2CRANCNFSM4HKYTEAQ> 
.
[ { "@context": "http://schema.org", "@type": "EmailMessage", 
"potentialAction": { "@type": "ViewAction", "target": 
"https://github.com/XCSoar/XCSoar/pull/190?email_source=notifications\u0026email_token=AL7IIITM4U6E43VQU3ZR3CLPVW2CRA5CNFSM4HKYTEA2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODVS3KNY#issuecomment-493204791", 
"url": 
"https://github.com/XCSoar/XCSoar/pull/190?email_source=notifications\u0026email_token=AL7IIITM4U6E43VQU3ZR3CLPVW2CRA5CNFSM4HKYTEA2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODVS3KNY#issuecomment-493204791", 
"name": "View Pull Request" }, "description": "View this Pull Request on 
GitHub", "publisher": { "@type": "Organization", "name": "GitHub", "url": 
"https://github.com" } } ]
﻿

MaxKellermann commented:
I suggest we reopen this one as soon as those serious issues are fixed.

eku60 commented:
Hello Max, because I don't know how to repair the repository, I will delete the the fork eku60/XCSoar.
I sent my changes in a mail to you. In order to implement the driver for the Becker AR6201 radio there were two new files, src/Device/Driver/AR62xx.cpp and src/Device/Driver/AR62xx.hpp added, and the build/driver.mk and the src/Device/Register.cpp were changed.
In the moment, I don't find time to deal with it. I compiled my fork for TARGET=UNIX, PC and WIN64 and tested the driver with a Becker AR6201 attached to a serial port on a Kubuntu and two Windows-systems. It worked on each system.
I know that many people are waiting for this driver, if you like to implement it into XCSoar, I can send them to you again, no problem.
Please let me know, whether or not you will do so.

Kind regards
Elmar

MaxKellermann commented:
If I have to clean up after you, then your work is not helpful for the XCSoar project. It doesn't matter how many people are waiting for the feature - if you're not doing it properly, your code is useless. Learn how to clean up. If you are incapable or unwilling to learn, don't expect anybody else to do it for you.

lordfolken commented:
Elmar, please understand we need maintainable software. A contribution is 33% of the work. It needs to integrate properly in the code base and have a reproduce able history.  Without that it gets very hard to debug/attribute failures / errors. In addition sub-optimal solutions will generate technological debts, that will eat up any progress due to requiring big time budgets to correct.

I understand Max has a harsh way of expressing things and this might seem offensive. But please realize that he has spent 1000s of hours (literally) just cleaning up and refactoring xcsoar's code. Its tedious work that has seemingly no immediate gain, but it really pays of in the long run, especially in the reduction of complexity.

Please copy your changes from your code base, checkout master newly and add those files back again. Try to split into as many micro-commits as possible.Then do another pull request.

eku60 commented:
Hello Philipp,

 

I really can understand you and I was thinking twice, whether or not my answer to you has more than my two cents to Max Kellermann and how to deal with.

 

The situation I had was a bit different. I spent many many days to get a PC with Kubuntu ready to compile XCSoar. Then I copied the KRT2 driver and changed the necessary methods for the use with the Becker radio. The methods are lend from the LK8000 Becker-driver and implemented in a very simple way. I soldered a cable (Becker has a RS422 interface), bought a converter RS232 <->RS422, connected the Becker to my PCs (one with win10, one with Kubuntu 18.10 and one with Ubuntu 18.10) respectively. XCSoar worked with my driver, the frequencies changed the way it should be.

 

Then I made a mistake with GIT in Eclipse (and that a fork is needed to perform a pull request, when I'm right). It was my fault and I didn`t know how to repair it in an easy way (that was the main problem). Then after some trials I got the 720 pages-protocol and found that some helpful, but not necessary methods (performing a log entry) were producing error messages (but not in my binary, I don't know why). I commented them out and tested again, driver is still working.

 

Today, I'm still not able to compile XCSoar for the Cubieboard (OpenVario). TARGET = CUBIE is not working, no one wanted or could give any help.

 

Philipp, I do that in my leisure time, I think as you do. Besides this, I'm president of our club, flight instructor and aircraft mechanic. That takes a lot of time too. In the mean time, I changed the ino-Sketchfile  for Stefly's dual rotary encoder and had some questions. No one was so unfriendly.

 

The answer of Max Kellermann is by far not acceptable! That is not the way to talk to others, who only want to help keeping this project running. I told that to him and thought it would be easy for him to evaluate the code and implement it into XCSoar. It is not large and I found it not difficult to understand. Even if he gives the code to someone else he knows and trusts, to bring it to an acceptable state. I only wanted to be able to use the driver, and I know many are waiting for it.

That's not the way to work with someone, who is rather unknown. Max doesn't know anything from me except the small more or less sufficient code and my trial to explain my problems.

 

People from the LK-development or around the OpenVario development gave more help, more understanding for my situation and are by far more friendly. Martin Heide and Peter Kremer from AS helped implementing the OpenVarios into our gliders and get them running.

 

I don't know how to deal with and whether or not I will spend time again to get the driver in the official XCSoar version. In the moment I'm very disappointed.

 

Regards

Elmar

 

 

Von: Philipp Wollschlegel [mailto:notifications@github.com] 
Gesendet: Mittwoch, 31. Juli 2019 17:35
An: XCSoar/XCSoar
Cc: eku60; Assign
Betreff: Re: [XCSoar/XCSoar] Driver for Becker AR62xx radios (#190)

 

Elmar, please understand we need maintainable software. A contribution is 33% of the work. It needs to integrate properly in the code base and have a reproduce able history. Without that it gets very hard to debug/attribute failures / errors. In addition sub-optimal solutions will generate technological debts, that will eat up any progress due to requiring big time budgets to correct.

I understand Max has a harsh way of expressing things and this might seem offensive. But please realize that he has spent 1000s of hours (literally) just cleaning up and refactoring xcsoar's code. Its tedious work that has seemingly no immediate gain, but it really pays of in the long run, especially in the reduction of complexity.

Please copy your changes from your code base, checkout master newly and add those files back again. Try to split into as many micro-commits as possible.Then do another pull request.

—
You are receiving this because you were assigned.
Reply to this email directly, view it on GitHub <https://github.com/XCSoar/XCSoar/pull/190?email_source=notifications&email_token=AL7IIIUTTIASPMYXFTDWPITQCGWJPA5CNFSM4HKYTEA2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD3HUZNY#issuecomment-516902071> , or mute the thread <https://github.com/notifications/unsubscribe-auth/AL7IIIQAFMFNMBQG5ZITOG3QCGWJPANCNFSM4HKYTEAQ> .  <https://github.com/notifications/beacon/AL7IIIUIVKYAKENAOGCACJ3QCGWJPA5CNFSM4HKYTEA2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD3HUZNY.gif> 



---
Diese E-Mail wurde von Avast Antivirus-Software auf Viren geprüft.
https://www.avast.com/antivirus

lordfolken commented:
I fully understand your position. I understand max, but i know he doesn't communicate diplomatically. However max will accept a contribution regardless of previous history, as long as its well done.

So how about we try to fix this mess a bit:

1. You don't seem to have a repository on github with your changes, that makes it a bit difficult to pull/review/request stuff:
To fix: 
  1. login to github
  2. goto https://github.com/XCSoar/XCSoar 
  3. press fork and put into your account
  4. clone the repository locally: goto your repo and get the clone url from the green button "clone or download"
  5. git clone <url>
  5a. cd into the directory
  6. (create a branch) git branch ar62xx_radio
  7. checkout the branch: git checkout ar62xx radio
  8. (add the hpp/cpp files) git add <path to file>
  9. commit those with a git commit 
 10. do the build.mk changes, do again git add <files>
 11. don't forgett to change the news.txt file
 12. add those changes as well with a commit. 
 13. git push (it will complain that the branch does not exist on the remote, do whatever it says)
 14. change to the main project xcsoar/xcsoar in the browser and press pull request
 15. select your repository

eku60 commented:
Hello Philipp,

 

I did all of the points up to point 14. At point 11. , I also added the changes to Register.cpp the same way as to driver.mk.

Just the git push is not yet done.

I will perform a local make run before the git push. I hope that all is correct, this time.

 

regards

eku60

Von: Philipp Wollschlegel [mailto:notifications@github.com] 
Gesendet: Freitag, 2. August 2019 13:05
An: XCSoar/XCSoar
Cc: eku60; Assign
Betreff: Re: [XCSoar/XCSoar] Driver for Becker AR62xx radios (#190)

 

I fully understand your position. I understand max, but i know he doesn't communicate diplomatically. However max will accept a contribution regardless of previous history, as long as its well done.

So how about we try to fix this mess a bit:

1.	You don't seem to have a repository on github with your changes, that makes it a bit difficult to pull/review/request stuff:
To fix:
2.	login to github
3.	goto https://github.com/XCSoar/XCSoar
4.	press fork and put into your account
5.	clone the repository locally: goto your repo and get the clone url from the green button "clone or download"
6.	git clone 
5a. cd into the directory
7.	(create a branch) git branch ar62xx_radio
8.	checkout the branch: git checkout ar62xx radio
9.	(add the hpp/cpp files) git add 
10.	commit those with a git commit
11.	do the build.mk changes, do again git add 
12.	don't forgett to change the news.txt file
13.	add those changes as well with a commit.
14.	git push (it will complain that the branch does not exist on the remote, do whatever it says)
15.	change to the main project xcsoar/xcsoar in the browser and press pull request
16.	select your repository

—
You are receiving this because you were assigned.
Reply to this email directly, view it on GitHub <https://github.com/XCSoar/XCSoar/pull/190?email_source=notifications&email_token=AL7IIIRV4RWVNJVNNBDKDRDQCQIEPA5CNFSM4HKYTEA2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD3NOAIY#issuecomment-517660707> , or mute the thread <https://github.com/notifications/unsubscribe-auth/AL7IIIQXJUVSVD6DHZ4HJZTQCQIEPANCNFSM4HKYTEAQ> .  <https://github.com/notifications/beacon/AL7IIIVNLDHD2WRZWSD44G3QCQIEPA5CNFSM4HKYTEA2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD3NOAIY.gif>

Werner86 commented:
Trying to get this topic back into code by https://github.com/XCSoar/XCSoar/pull/449

eku60 commented:
Hello Werner86,

 

can you send me an email to elmar.kuemmel@t-online.de with your private address?

 

regards

eku60

 

Von: Werner86 [mailto:notifications@github.com] 
Gesendet: Dienstag, 7. Juli 2020 15:35
An: XCSoar/XCSoar
Cc: eku60; Assign
Betreff: Re: [XCSoar/XCSoar] Driver for Becker AR62xx radios (#190)

 

Trying to get this topic back into code by #449 <https://github.com/XCSoar/XCSoar/pull/449> 

—
You are receiving this because you were assigned.
Reply to this email directly, view it on GitHub <https://github.com/XCSoar/XCSoar/pull/190#issuecomment-654863903> , or unsubscribe <https://github.com/notifications/unsubscribe-auth/AL7IIIVH4ZCUC66FB7DTXXLR2MP73ANCNFSM4HKYTEAQ> .  <https://github.com/notifications/beacon/AL7IIIXEDTWEV4342CIY7MDR2MP73A5CNFSM4HKYTEA2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOE4EG4HY.gif>

Werner86 commented:
done :-)
